### PR TITLE
Update serverless docs with newer architecture

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- update serverless docs:
  - remove mentions to KService and instead describe usage of deployments, services and hpa
  - update svgs 
- update few dead links in event-mesh docs

**Related issue(s)**

See https://github.com/kyma-project/kyma/issues/8356

### CODE REVIEW

File: docs/serverless/03-01-supported-webhooks.md [hunk 1]
```diff
@@ -29,4 +29,4 @@ A newly created or modified Function CR is first updated by the defaulting webho
     - Requests are lower than or equal to limits, and the minimum number of replicas is lower than or equal to the maximum one.
     - The Function CR contains all the required parameters.
     - The format of deps, envs, labels, and the Function name ([RFC 1035](https://tools.ietf.org/html/rfc1035)) is correct.
-    - The Function CR contains any envs reserved for the KService: `K_REVISION`, `K_CONFIGURATION`, `K_SERVICE`, `FUNC_RUNTIME`, `FUNC_HANDLER`, `FUNC_TIMEOUT`, `FUNC_PORT`, `PORT`, `MOD_NAME`, `NODE_PATH`.
+    - The Function CR contains any envs reserved for the Deployment: `FUNC_RUNTIME`, `FUNC_HANDLER`, `FUNC_TIMEOUT`, `FUNC_PORT`, `PORT`, `MOD_NAME`, `NODE_PATH`.
```
Comment (aerfio) [inline docs/serverless/03-01-supported-webhooks.md:32]: note to self -> there are two more, implementation in separate PR

File: docs/serverless/06-01-function-cr.md [hunk 2]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| **status.conditions.status**             | Not applicable | Describes the status of processing the Function CR by the Function Controller. It can be `True` for success, `False` for failure, or `Unknown` if the CR processing is still in progress. If the status of all conditions is `True`, the overall status of the Function CR is ready.                                                                                                                                                                                                                                              |
+| **status.conditions.type**               | Not applicable | Describes a substage of the Function CR processing. There are three condition types that a Function has to meet to be ready: `ConfigurationReady`, `BuildReady`, and `Running`. When displaying the Function status in the terminal, these types are shown under `CONFIGURED`, `BUILT`, and `RUNNING` columns respectively. All condition types can change asynchronously depending on the type of Function modification, but all three need to be in the `True` status for the Function to be considered successfully processed. |
+
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:97]: This part is missing:

```
### Status reasons

Processing of a Function CR can succeed, continue, or fail for one of these reasons:
```

File: docs/serverless/06-01-function-cr.md [hunk 3]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `JobCreated`                     | `BuildReady`         | The Kubernetes Job resource that builds the Function image was created.                                                   |
+| `JobRunning`                     | `BuildReady`         | The Job is in progress.                                                                                                   |
+| `JobsDeleted`                    | `BuildReady`         | Previous Jobs responsible for building Function images were deleted.                                                      |
+| `JobFinished`                    | `BuildReady`         | The Job was finished and the Function's image was uploaded to the Docker Registry.                                        |
+| `DeploymentCreated`              | `Running`            | A new Deployment referencing the Function's image was created.                                                            |
+| `DeploymentUpdated`              | `Running`            | The existing Deployment was updated after such changes as the Function's image, scaling parameters, variables, or labels. |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:109]:
```suggestion
| `DeploymentUpdated`              | `Running`            | The existing Deployment was updated after changing the Function's image, scaling parameters, variables, or labels. |
```

Comment (kazydek) [reply docs/serverless/06-01-function-cr.md:109]: (I know it was already there, but looking at it now, I would put it this way ;) )

File: docs/serverless/06-01-function-cr.md [hunk 4]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `DeploymentFailed`               | `Running`            | The Function's Pod crashed or could not start due to an error.                                                            |
+| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for Deployment's readiness.                                                      |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:111]:
```suggestion
| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for the Deployment to be ready.                                                      |
```

File: docs/serverless/06-01-function-cr.md [hunk 5]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `DeploymentReady`                | `Running`            | The Function was deployed and ready to serve.                                                                             |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:112]:
```suggestion
| `DeploymentReady`                | `Running`            | The Function was deployed and is ready.                                                                             |
```

Comment (kazydek) [reply docs/serverless/06-01-function-cr.md:112]: "ready to serve" - you mean like pasta? ;) If you want to use "serve," you would rather need to explain what it serves like "deployment is ready to serve the traffic". Still, I wouldn't use it, unless you insist ;)

File: docs/serverless/06-01-function-cr.md [hunk 6]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `ServiceCreated`                 | `Running`            | A new Service referencing the Function's Deployment was created.                                                          |
+| `ServiceUpdated`                 | `Running`            | The existing Service was updated after needed changes.                                                                    |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:114]:
```suggestion
| `ServiceUpdated`                 | `Running`            | The existing Service was updated after applying required changes.                                                                    |
```

Comment (kazydek) [reply docs/serverless/06-01-function-cr.md:114]: (definitely more common)

File: docs/serverless/06-01-function-cr.md [hunk 7]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `ServiceError`                   | `Running`            | There are some problems with Kubernetes Service.                                                                         |
+| `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's deployment was created.                                              |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:116]:
```suggestion
| `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's Deployment was created.                                              |
```

File: docs/serverless/06-01-function-cr.md [hunk 8]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after needed changes.                                                        |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:117]:
```suggestion
| `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after applying required changes.                                                        |
```

File: docs/serverless/06-01-function-cr.md [hunk 9]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `HorizontalPodAutoscalerError`   | `Running`            | There are some problems with HorizontalPodAutoscaler.                                                                     |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:118]:
```
The HorizontalPodAutoscaler could not be created or updated due to an error.                                                          
```

File: docs/serverless/06-01-function-cr.md [hunk 10]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| `DeploymentFailed`               | `Running`            | The Function's Pod crashed or could not start due to an error.                                                            |
+| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for Deployment's readiness.                                                      |
+| `DeploymentReady`                | `Running`            | The Function was deployed and ready to serve.                                                                             |
+| `ServiceCreated`                 | `Running`            | A new Service referencing the Function's Deployment was created.                                                          |
+| `ServiceUpdated`                 | `Running`            | The existing Service was updated after needed changes.                                                                    |
+| `ServiceError`                   | `Running`            | There are some problems with Kubernetes Service.                                                                         |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:115]: Yeah, that one is quite vague. So there is no particular reason but a given (human-readable) error pops up? Maybe sth like this:

```
The Kubernetes Service could not be created or updated due to an error.                                                          
```

File: docs/serverless/06-01-function-cr.md [hunk 11]
```diff
@@ -75,57 +75,62 @@ spec:
...
-The Function custom resource relies on these Kubernetes and [Knative Serving](https://knative.dev/docs/serving/) resources:
+The Function custom resource relies on these Kubernetes resources:
 
-| Resource | Description |
-|----------|-------------|
-|[KService CR](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md#service) | Orchestrates the deployment and availability of the Function.|
-|[Kubernetes Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) | Builds an image with the Function code on a runtime. |
+| Resource                                                                                              | Description                                                                   |
+| ----------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------- |
+| [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)                             | Stores the source code and dependencies of the Function.                      |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:126]:
```suggestion
| [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)                             | Stores the Function's source code and dependencies.                      |
```

File: docs/serverless/06-01-function-cr.md [hunk 12]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| [Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)              | Builds an image with the Function code in a runtime.                          |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:127]:
```suggestion
| [Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)              | Builds an image with the Function's code in a runtime.                          |
```

File: docs/serverless/06-01-function-cr.md [hunk 13]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)                   | Serves the Function image as microservice.                                    |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:128]:
```suggestion
| [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)                   | Serves the Function's image as a microservice.                                    |
```

File: docs/serverless/06-01-function-cr.md [hunk 14]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| [Service](https://kubernetes.io/docs/concepts/services-networking/service/)                           | Exposes Function's Deployment as a network service inside Kubernetes cluster. |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:129]:
```suggestion
| [Service](https://kubernetes.io/docs/concepts/services-networking/service/)                           | Exposes the Function's Deployment as a network service inside the Kubernetes cluster. |
```

File: docs/serverless/06-01-function-cr.md [hunk 15]
```diff
@@ -75,57 +75,62 @@ spec:
...
+| [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | Automatically scales the number of Function's pods.                           |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:130]:
```suggestion
| [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | Automatically scales the number of Function's Pods.                           |
```

File: docs/serverless/02-01-serverless.md [hunk 16]
```diff
@@ -2,44 +2,32 @@
 title: Architecture
 ---
 
-Serverless relies on [Knative Serving](https://knative.dev/docs/serving/) for deploying and managing Functions and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. See how these and other resources process a Function within a Kyma cluster:
+Serverless relies on Kubernetes resources like [Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), [Services](https://kubernetes.io/docs/concepts/services-networking/service/) and [HorizontalPodAutoscalers](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) for deploying and managing Functions and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) for creating Docker images. See how these and other resources process a Function within a Kyma cluster:
 
-![Serverless architecture](./assets/serverless-architecture.svg)
+ <!-- note: we are using also svc and hpa, maybe a list would be suitable here? -->
```
Comment (aerfio) [inline docs/serverless/02-01-serverless.md:7]:
```suggestion
```

[the remark this answers is not in the document]
Comment (aerfio) [reply docs/serverless/02-01-serverless.md:7]: I've thought about making part line 5 a list, but we've done it the other way, so I'm resolving this comment

File: docs/serverless/03-04-function-processing.md [hunk 17]
```diff
@@ -32,14 +32,18 @@ This initial phase starts when you create a Function CR with configuration speci
...
+This stage revolves around creating a Deployment, Service and HorizontalPodAutoscaler or updating them when configuration changes were made in the Function CR or the Function image was rebuilt.
+
+In general, the Deployment is considered updated when both configuration and the image tag in the Deployment are up to date. Service and HorizontalPodAutoscaler are considered updated, when there are proper labels set and configuration is up to date.
+
+Thanks to the implemented reconciliation loop, the Function Controller constantly observes all newly created or updated resources. If it detects changes, it fetches the appropriate resource's status and only then updates the Function's status.
 
 ![Function running](./assets/running.svg)
```
[the remark this answers is not in the document]
Comment (aerfio) [reply docs/serverless/03-04-function-processing.md:49]: yup, updated diagram!

File: docs/serverless/06-01-function-cr.md (hunk 9 above)
Comment (aerfio) [reply docs/serverless/06-01-function-cr.md]: Those problems are more connected with some incorrect config, not with creation

File: docs/serverless/06-01-function-cr.md (hunk 10 above)
Comment (aerfio) [reply docs/serverless/06-01-function-cr.md]: Same answer as with HPA, I'll add some example as to what can go wrong with them

File: docs/serverless/02-01-serverless.md [hunk 18]
```diff
@@ -2,44 +2,32 @@
...
+
+![Serverless architecture](./assets/serverless-architecture.svg) <!-- needs update -->
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/serverless/02-01-serverless.md:8]: Really nice this time :) I can see just one tiny into a 90 degree angle ;)
image

File: docs/serverless/06-01-function-cr.md [hunk 19]
```diff
@@ -75,57 +75,66 @@ spec:
...
+| `DeploymentFailed`               | `Running`            | The Function's Pod crashed or could not start due to an error.                                                     |
+| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for the Deployment to be ready.                                           |
+| `DeploymentReady`                | `Running`            | The Function was deployed and is ready.                                                                            |
+| `ServiceCreated`                 | `Running`            | A new Service referencing the Function's Deployment was created.                                                   |
+| `ServiceUpdated`                 | `Running`            | The existing Service was updated after applying required changes.                                                  |
+| `ServiceError`                   | `Running`            | There are some problems with Kubernetes Service.                                                                   |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:119]:
```suggestion
| `ServiceError`                   | `Running`            | There are issues with the Kubernetes Service that relate to configuration.                                                                   |
```

File: docs/serverless/06-01-function-cr.md [hunk 20]
```diff
@@ -75,57 +75,66 @@ spec:
...
+| `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's Deployment was created.                                       |
+| `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after applying required changes.                                      |
+| `HorizontalPodAutoscalerError`   | `Running`            | There are some problems with HorizontalPodAutoscaler.                                                              |
```
Comment (kazydek) [inline docs/serverless/06-01-function-cr.md:122]:
```suggestion
| `HorizontalPodAutoscalerError`   | `Running`            | There are issues with the HorizontalPodAutoscaler that relate to configuration.                                                           |
```

File: docs/serverless/assets/serverless-architecture.svg [hunk 21]
```diff
@@ -1,3 +1,3 @@
...
...all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 14px; height: 1px; padding-top: 455px; margin-left: 616px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 15px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 17px" color="#ffffff">10</font></div></div></div></foreignObject><text x="623" y="460" fill="#000000" font-family="Helvetica" font-size="15px" text-anchor="middle">10</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://desk.draw.io/support/solutions/articles/16000042487" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
[the remark this answers is not in the document]
Comment (aerfio) [reply docs/serverless/assets/serverless-architecture.svg]: Done

File: docs/serverless/03-01-supported-webhooks.md [hunk 22]
```diff
@@ -7,26 +7,26 @@ A newly created or modified Function CR is first updated by the defaulting webho
...
+   | **minReplicasValue** | `1`            |
 
-    - Requests are lower than or equal to limits, and the minimum number of replicas is lower than or equal to the maximum one.
-    - The Function CR contains all the required parameters.
-    - The format of deps, envs, labels, and the Function name ([RFC 1035](https://tools.ietf.org/html/rfc1035)) is correct.
-    - The Function CR contains any envs reserved for the KService: `K_REVISION`, `K_CONFIGURATION`, `K_SERVICE`, `FUNC_RUNTIME`, `FUNC_HANDLER`, `FUNC_TIMEOUT`, `FUNC_PORT`, `PORT`, `MOD_NAME`, `NODE_PATH`.
+   - Requests are lower than or equal to limits, and the minimum number of replicas is lower than or equal to the maximum one.
+   - The Function CR contains all the required parameters.
+   - The format of deps, envs, labels, and the Function name ([RFC 1035](https://tools.ietf.org/html/rfc1035)) is correct.
+   - The Function CR contains any envs reserved for the Deployment: `FUNC_RUNTIME`, `FUNC_HANDLER`, `FUNC_TIMEOUT`, `FUNC_PORT`, `PORT`, `MOD_NAME`, `NODE_PATH`, `REQ_MB_LIMIT`, `FUNC_MEMORY_LIMIT`.
```
Comment (michal-hudy) [inline docs/serverless/03-01-supported-webhooks.md:32]: `PORT` was reserved by knative service

**Pull request merged**